### ISSUE TITLE
Nested query conditions in repository.remove cause TypeError

### ISSUE THREAD

**Issue by jasperroloff** [mikro-orm/mikro-orm#492]
**Describe the bug**
When calling the `.remove(...)` method of a repository with nested query conditions, it causes a TypeError.

**Stack trace**
```
(node:66382) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'slice' of undefined
    at ObjectCriteriaNode.autoJoin (PROJECT_DIR/node_modules/mikro-orm/dist/query/CriteriaNode.js:215:37)
    at ObjectCriteriaNode.process (PROJECT_DIR/node_modules/mikro-orm/dist/query/CriteriaNode.js:168:26)
    at PROJECT_DIR/node_modules/mikro-orm/dist/query/CriteriaNode.js:172:39
    at Array.reduce (<anonymous>)
    at ObjectCriteriaNode.process (PROJECT_DIR/node_modules/mikro-orm/dist/query/CriteriaNode.js:170:42)
    at QueryBuilder.init (PROJECT_DIR/node_modules/mikro-orm/dist/query/QueryBuilder.js:274:99)
    at QueryBuilder.delete (PROJECT_DIR/node_modules/mikro-orm/dist/query/QueryBuilder.js:54:21)
    at MariaDbDriver.nativeDelete (PROJECT_DIR/node_modules/mikro-orm/dist/drivers/AbstractSqlDriver.js:95:69)
    at EntityManager.nativeDelete (PROJECT_DIR/node_modules/mikro-orm/dist/EntityManager.js:186:39)
    at EntityManager.remove (PROJECT_DIR/node_modules/mikro-orm/dist/EntityManager.js:297:21)
    at EntityRepository.remove (PROJECT_DIR/node_modules/mikro-orm/dist/entity/EntityRepository.js:36:24)
    at PROJECT_DIR/src/main.ts:33:47
    at Generator.next (<anonymous>)
    at fulfilled (PROJECT_DIR/src/main.ts:4:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:66382) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
(node:66382) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**To Reproduce**
I've created a sample repository to reproduce this issue, use `ts-node src/main.ts` to run.
https://github.com/jasperroloff/mikro-orm-nested-query-delete-issue

**Expected behavior**
It should successfully delete the entity, since a findOne given the same query does succeed.

**Additional context**
I've experienced this issue with (at least) the mariadb driver and the ReflectMetadataProvider.

**Versions**

| Dependency | Version |
| - | - |
| node | v13.13.0 |
| typescript | ^3.8.3 |
| mikro-orm | ^3.6.8 |
| mariadb | ^2.3.1 |
| mysql2 | ^2.1.0 |
| ts-node | ^8.8.2 |

(see package.json in sample repo)

**Comment by B4nan** [comment]
This was never meant for update or delete queries. Will see if I can make this work in v4. 

As a workaround, select the entity as usual and remove the reference you get.

**Comment by jasperroloff** [comment]
Thanks for your reply, I thought MikroORM is supporting this already and I just found a bug.
The workaround you mentioned, I am using already.

So in this case, consider this more as a feature request than a bug 😄

**Comment by B4nan** [comment]
Closing as implemented via #538, which is now merged in v4 (`dev` branch). Subscribe here for updates #527.
